sbi/samplers/rejection/rejection.py

```diff
@@ -283,7 +306,19 @@ def accept_reject_sample(
     num_sampled_total = torch.zeros(num_xos)
     num_samples_possible = 0
 
+    start_time = time.time()
     while num_remaining > 0:
+        if (
+            max_sampling_time is not None
+            and (time.time() - start_time) > max_sampling_time
+        ):
+            raise RuntimeError(
+                "Rejection sampling exceeded max_sampling_time."
+                "Sampling aborted early"
+                "Because of extremely low acceptance."
+                "Consider switching to 'sample_with=mcmc'."
```

Comment:
```suggestion
                    "Sampling aborted early because rejection sampling exceeded "
                    "max_sampling_time. This is likely due to extremely low "
                    "acceptance. Consider switching to MCMC or VI, or checking "
                    "for model misspecification."
```